services/controller/standard/events.go

```diff
@@ -332,3 +339,61 @@ func (s *Service) subscribeToBeaconCommittees(ctx context.Context,
 	s.subscriptionInfos[epoch] = subscriptionInfo
 	s.subscriptionInfosMutex.Unlock()
 }
+
+// VerifySyncCommitteeMessages handles the "head" events from the beacon node for sync committee validation.
+func (s *Service) VerifySyncCommitteeMessages(ctx context.Context, data *apiv1.HeadEvent) {
+	_, span := otel.Tracer("attestantio.vouch.services.controller.standard").Start(ctx, "VerifySyncCommitteeMessages")
+	defer span.End()
+
+	// We verify against the previous slot as that is when the sync committee will have reported.
+	previousSlot := data.Slot - 1
+	currentSlot := data.Slot
+
+	// Logging with the current slot as that is when this code is executed.
+	log := log.With().
+		Uint64("current_slot", uint64(currentSlot)).
+		Uint64("previous_slot", uint64(previousSlot)).
+		Str("block_header", data.Block.String()).
+		Logger()
+	log.Trace().Msg("Received head event")
+
+	previousSlotData, found := s.syncCommitteeMessenger.GetDataUsedForSlot(previousSlot)
+	if !found {
+		log.Trace().Msg("No reported sync committee message data for slot; skipping validation")
+		return
+	}
+	blockResponse, err := s.signedBeaconBlockProvider.SignedBeaconBlock(ctx, &api.SignedBeaconBlockOpts{
+		Block: data.Block.String(),
+	})
+	if err != nil {
+		log.Trace().Err(err).Msg("Failed to retrieve head block for sync committee validation")
+		return
+	}
+	parentRoot, err := blockResponse.Data.ParentRoot()
+	if err != nil {
+		log.Trace().Err(err).Msg("Failed to get parent root of head block")
+		return
+	}
+	if parentRoot.String() != previousSlotData.Root.String() {
+		log.Trace().Str("head_parent_root", parentRoot.String()).Str("broadcast_root", previousSlotData.Root.String()).
```

Comment:
Rather than `Str("head_parent_root", parentRoot.String())`  you can use `Stringer("head_parent_root", parentRoot)` which calls the `String()` function internally.